pkg/controller/persistentvolume/persistentvolume_recycler_controller.go

```diff
@@ -40,21 +40,39 @@ var _ volume.VolumeHost = &PersistentVolumeRecycler{}
 // This controller will Recycle those volumes whose reclaim policy is set to PersistentVolumeReclaimRecycle and make them
 // available again for a new claim.
 type PersistentVolumeRecycler struct {
-	volumeController *framework.Controller
-	stopChannel      chan struct{}
-	client           recyclerClient
-	kubeClient       client.Interface
-	pluginMgr        volume.VolumePluginMgr
-	cloud            cloudprovider.Interface
+	volumeController  *framework.Controller
+	stopChannel       chan struct{}
+	client            recyclerClient
+	kubeClient        client.Interface
+	pluginMgr         volume.VolumePluginMgr
+	cloud             cloudprovider.Interface
+	maximumRetryCount int
+	syncPeriod        time.Duration
+	// Local cache of failed recycle / delete operations. Map volume.Name -> status of the volume.
+	// Only PVs in Failed state have an entry here.
+	failedVolumes map[string]failedVolumeStatus
+}
+
+// failedVolumeStatus holds state of failed delete/recycle operation on a volume.
+// The controller re-tries the opration several times and it stores retry
+// count + timestamp of the last attempt here.
+type failedVolumeStatus struct {
+	// How many recycle/delete operations failed.
+	retryCount int
+	// Timestamp of the last attempt.
+	lastAttempt time.Time
 }
 
 // PersistentVolumeRecycler creates a new PersistentVolumeRecycler
```

Comment:
this is my typo in the original, but since you've got this PR open, would you fix the comment to start with the correct func name?